### PR TITLE
Remove IManagedPointerTypeReference

### LINKED ISSUE #14027: Remove IManagedPointerTypeReference interface
`IManagedPointerTypeReference` is not implemented and can be removed.

### PR DESCRIPTION
**Pull request by TyOverby**
Closes #14027

### CODE REVIEW

**Comment by TyOverby** [comment]
@cston: As I understand, this is a change required from one of your earlier commits.  Please take a look and see if it meets your expectations.

**Comment by TyOverby** [comment]
Somehow `CompilersBoundTreeGenerator.csproj` is failing to build, but I can't repro on my own machine.

File: src/Compilers/Core/Portable/PEWriter/TypeNameSerializer.cs [hunk 1]
```diff
@@ -56,30 +56,6 @@ internal static string GetSerializedTypeName(this ITypeReference typeReference,
...
-            {
-                typeReference = reference.GetTargetType(context);
-                bool isAssemQual = false;
-                AppendSerializedTypeName(sb, typeReference, ref isAssemQual, context);
-                sb.Append('&');
-                goto done;
-            }
-
-            INamespaceTypeReference namespaceType = typeReference.AsNamespaceTypeReference;
-            if (namespaceType != null)
```
Comment (cston) [inline src/Compilers/Core/Portable/PEWriter/TypeNameSerializer.cs]: Is this case not needed any more?

Comment (TyOverby) [inline src/Compilers/Core/Portable/PEWriter/TypeNameSerializer.cs]: Overzealous line delete :|

**Comment by cston** [comment]
LGTM

**Comment by jaredpar** [comment]
Sad we missed merging this.  Please re-target to post-dev15 so we can merge.

File: src/Compilers/Core/Portable/PEWriter/TypeNameSerializer.cs [hunk 2]
```diff
@@ -80,6 +70,7 @@ internal static string GetSerializedTypeName(this ITypeReference typeReference,
                 goto done;
             }
 
+
```
Comment (jcouv) [inline src/Compilers/Core/Portable/PEWriter/TypeNameSerializer.cs]: nit: extra line

**Comment by TyOverby** [comment]
@jaredpar @cston 

I've retargeted for post-dev15

**Comment by jaredpar** [comment]
retest eta please

**Comment by TyOverby** [comment]
I rebased again because the RepoUtil bug didn't go away.

**Pull request merged**